File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        vector<vector<string>> op;
        unordered_map<string, vector<string>> mp;

        for(int i=0;i<strs.size();i++)
        {
            string currStr = strs[i];
            sort(currStr.begin(), currStr.end());
            
            mp[currStr].push_back(strs[i]);
        }

        for(auto it:mp)
        {
            op.push_back(it.second);
        }

        return op;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        dictt = {}

        for i in strs:
            ss = "".join(sorted(i))
            if ss in dictt:
                dictt[ss].append(i)
            else:
                dictt[ss] = [i]

        op = []
        for i in dictt:
            op.append(dictt[i])
        return op
        

        